internal/bundle/plain_test.go

```diff
@@ -0,0 +1,124 @@
+		mockfs := fstest.MapFS{
+			"manifests/invalid.yaml": {
+				Data: []byte(""),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("required to contain at least one object"))
+	})
+	It("should fail if any of the files are invalid json or yaml", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
+			"manifests/invalid.yaml": {
+				Data: []byte("not;valid"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("cannot unmarshal string into Go value"))
+	})
+	It("should fail if the YAML or JSON objects do not specify `kind`", func() {
+		mockfs := fstest.MapFS{
+			"manifests/invalid.yaml": {
+				Data: []byte("almost: valid"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("Object 'Kind' is missing"))
+	})
+	It("should pass if the YAML is valid and contains a `kind` element", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err).To(BeNil())
+	})
+	It("should pass if the JSON is valid and contains a `kind` element", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid-json.yaml": {
+				Data: []byte(`{"kind": "ofnecessary"}`),
+			},
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
+		}
+		plainv0, err := LoadPlainV0(mockfs)
+		Expect(len(plainv0.Objects)).To(Equal(2))
+		Expect(err).To(BeNil())
+	})
+	It("should pass if some manifests are JSON and some are YAML", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid-json.yaml": {
+				Data: []byte(`{"kind": "ofnecessary"}`),
+			},
+			"manifests/valid-yaml.yaml": {
+				Data: []byte(`kind: ofnecessary`),
+			},
+		}
+		plainv0, err := LoadPlainV0(mockfs)
+		Expect(len(plainv0.Objects)).To(Equal(2))
+		Expect(err).To(BeNil())
+	})
+
+	It("should fail if there are nested subdirectories", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary"),
+			},
+			"manifests/invalid/valid.yaml": {
+				Data: []byte("kind: illegalsubdir"),
+			},
+		}
+		_, err := LoadPlainV0(mockfs)
+		Expect(err.Error()).To(ContainSubstring("subdirectories are not allowed"))
+	})
+	It("should pass with multi-object yaml files", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: ofnecessary\n---\nkind: me"),
+			},
+		}
+		plainv0, err := LoadPlainV0(mockfs)
+		Expect(len(plainv0.Objects)).To(Equal(2))
+		Expect(err).To(BeNil())
+	})
+	It("should flatten and pass with metav1.List yaml files", func() {
+		mockfs := fstest.MapFS{
+			"manifests/valid.yaml": {
+				Data: []byte("kind: List\napiVersion: v1\nitems:\n  - kind: me\n    apiVersion: v1\n  - kind: you\n    apiVersion: v1"),
+			},
+		}
+		plainv0, err := LoadPlainV0(mockfs)
+		Expect(len(plainv0.Objects)).To(Equal(2))
+		Expect(err).To(BeNil())
+	})
```

Comment:
Another potential test: 2 files, one is valid YAML, the other is empty.

From the code that exists right now, that looks valid. But I wonder if we should call that invalid (i.e. every file in the manifests dir MUST have at least one object)